### DISCUSSION TITLE
Installing Anaconda using brew

### DISCUSSION THREAD

**Discussion by duemme** [Homebrew/discussions#1332] [Getting started]
I'm really confused on how to use homebrew and anaconda togheter.
Reading online I understood that it is a bad idea to use brew and install anaconda using its default installer because brew and conda doesn't go well together.
So I would like installed anaconda using brew. At the end of the procedure I find to message: "==> Changing ownership of paths required by anaconda; your password may be necessary". What I'm supposed to do?
If I want to install a new python package (for example bs4) what should I use? conda? pip? Do I have to add the conda bin to my path variable?
If I add anaconda bin to my path variable and then I install python3.9 do I get any conflict?
If I want to install pycharm do I have to use brew or conda?
Where do I find documentation on how to clear this doubts?
Thanks

**Comment by danielnachun** [comment]
I have used Homebrew and Anaconda together for several years with no major issues.  Here are my recommendations:

1) Use Miniconda instead of the full Anaconda distribution.  This keeps things a lot simpler
2) The first command you should run after installing Miniconda is `conda config --set auto_activate_base false`.  This disables auto activation of the base environment (which you shouldn't really use for anything anyway) while still allowing you to use `conda activate` to activate other environments, and avoids the vast majority of problems that arise when using Homebrew and Miniconda together.
3) Do not try to use Miniconda as a general purpose package manager!  Instead think of it as a way to install specific environments which you activate when you need them, and deactivate when done.  Generally speaking I try to use Homebrew or BrewSci (https://github.com/brewsci) to install tools whenever they are available there, and I use Miniconda only when the software is not available there.
4) Specifically in the case of Python environments, you should only use Miniconda to manage those.  Homebrew's Python should really be thought of more as a runtime like the JDK is for Java or the CLR for .NET.

**Comment by duemme** [comment]
Thank you Daniel,
meanwhile I found a solution that seems in line with what you wrote.
1 – I installed Anaconda with homebrew. The full Anaconda seems ok for me because I would like to use Jupiter Notebook that comes with Anaconda and I like Anaconda Navigator.
2 – I understood that everything had been installed in /usr/local/Anaconda without touching the PATH variable. So I go there and start the Anaconda Navigator when I need it (I’m thinking about creating a link and put the link in the App folder to find it with spotlight).
3 – with anaconda navigator I created a new environment that I’m using for learning how to code;
4 – I can use this environment with Spyder (I installed the spyder-kernels with anaconda navigator) or PyCharm (configured it to use this interpreter)
5 – I can also create new environments within PyCharm using conda.
6 – If I need additional packages I install them with anaconda navigator
7 – everything is separate and I’m pretty happy with this. Maybe the documentation on how to set everything (and why set it in a way or another) for a non-expert like me is a bit missing.

Thanks for your time

From: Daniel Nachun ***@***.***>
Date: Saturday, 24 April 2021 at 06:08
To: Homebrew/discussions ***@***.***>
Cc: duemme ***@***.***>, Author ***@***.***>
Subject: Re: [Homebrew/discussions] Installing Anaconda using brew (#1332)

I have used Homebrew and Anaconda together for several years with no major issues. Here are my recommendations:

  1.  Use Miniconda instead of the full Anaconda distribution. This keeps things a lot simpler
  2.  The first command you should run after installing Miniconda is conda config --set auto_activate_base false. This disables auto activation of the base environment (which you shouldn't really use for anything anyway) while still allowing you to use conda activate to activate other environments, and avoids the vast majority of problems that arise when using Homebrew and Miniconda together.
  3.  Do not try to use Miniconda as a general purpose package manager! Instead think of it as a way to install specific environments which you activate when you need them, and deactivate when done. Generally speaking I try to use Homebrew or BrewSci (https://github.com/brewsci) to install tools whenever they are available there, and I use Miniconda only when the software is not available there.
  4.  Specifically in the case of Python environments, you should only use Miniconda to manage those. Homebrew's Python should really be thought of more as a runtime like the JDK is for Java or the CLR for .NET.
